ui/ducks/confirm-transaction/confirm-transaction.duck.js

```diff
@@ -308,3 +344,38 @@ export function clearConfirmTransaction() {
     type: CLEAR_CONFIRM_TRANSACTION,
   };
 }
+
+export const updateTxValueIfMaxEthSettled = (
+  dispatch,
+  state,
+  { gasFeeEstimates, gasEstimateType },
+) => {
+  const txId = state.confirmTransaction?.txData?.id;
+
+  if (
+    txId &&
+    state.confirmTransaction.sendTxDetailPerId?.[txId]?.amountMode ===
+      AMOUNT_MODES.MAX &&
+    state.confirmTransaction.sendTxDetailPerId?.[txId]?.assetType ===
```

Comment:
Nit: An intermediate variable here would prevent repeating the long chaining and also serve as documentation:
```
const txId = state.confirmTransaction?.txData?.id;
const sendTxDetail = state.confirmTransaction?.sendTxDetailPerId?.[txId];
```

Reply:
This part is removed but thanks for the suggestion